static/app/components/profiling/flamegraph/flamegraphZoomView.tsx

```diff
@@ -366,21 +363,36 @@ function FlamegraphZoomView({
 
     const onResetZoom = () => {
       setConfigSpaceCursor(null);
+      setHoveredNode(null);
     };
 
     const onZoomIntoFrame = (frame: FlamegraphFrame, _strategy: 'min' | 'exact') => {
       if (frame) {
         selectedFramesRef.current = [frame];
       }
       setConfigSpaceCursor(null);
+      setHoveredNode(null);
+    };
+
+    const onHighlightFrame = (
+      frames: FlamegraphFrame[] | null,
+      type: 'hover' | 'selected'
+    ) => {
+      if (type === 'selected') {
+        selectedFramesRef.current = frames;
+      } else {
+        setHoveredNode(frames ? frames[0] : null);
```

Comment:
Should this check the length of frames as well? Otherwise, we could risk setting it to `undefined` which I'm not sure is handled correctly everywhere.

Reply:
Yup, fixed it by just making it frames[0] ?? null